e2e/suites/actions/version-actions.test.ts

```diff
@@ -0,0 +1,180 @@
+import { VersionManagePage } from '../../../projects/aca-testing-shared/src/components/version-manage/version-manager';
+import { Viewer } from '../../../projects/aca-testing-shared/src/components';
+
+describe('Version component actions', () => {
+  const versionManagePage = new VersionManagePage();
+  const viewerPage = new Viewer();
+
+  const username = `user-${Utils.random()}`;
+
+  let fileId: string;
+
+  const filesToUpload = [FILES.pdfFile, FILES.docxFile, FILES.xlsxFile, FILES.jpgFile, FILES.docxFile2, FILES.xlsxFile2];
+
+  const apis = {
+    admin: new RepoClient(),
+    user: new RepoClient(username, username)
+  };
+
+  const loginPage = new LoginPage();
+  const page = new BrowsingPage();
+  const { dataTable, toolbar } = page;
+  const uploadNewVersionDialog = new UploadNewVersionDialog();
+  const { searchInput } = page.header;
+
+  beforeAll(async (done) => {
+    await apis.admin.people.createUser({ username });
+    fileId = (await apis.user.upload.uploadFile(filesToUpload[0])).entry.id;
+    done();
+  });
+
+  afterAll(async (done) => {
+    await apis.user.nodes.deleteNodeById(fileId);
+    done();
+  });
+
+  describe('on Personal Files', () => {
+    beforeAll(async (done) => {
+      await apis.user.shared.shareFilesByIds([fileId]);
+      await loginPage.loginWith(username);
+      await page.clickPersonalFilesAndWait();
+
+      for (let i = 0; i < filesToUpload.length - 1; i++) {
+        await dataTable.selectItem(filesToUpload[i]);
+        await toolbar.clickMoreActionsUploadNewVersion();
+        await Utils.uploadFileNewVersion(filesToUpload[i + 1]);
+
+        await page.waitForDialog();
+
+        await uploadNewVersionDialog.majorOption.click();
+        await uploadNewVersionDialog.enterDescription('new major version description');
+        await uploadNewVersionDialog.uploadButton.click();
+        await uploadNewVersionDialog.waitForDialogToClose();
+      }
+      done();
+    });
+
+    it('[C586766] Should be possible to view a previous document version', async () => {
+      await dataTable.selectItem(filesToUpload[5]);
+      await toolbar.clickMoreActionsManageVersions();
+      await versionManagePage.viewFileVersion('1.0');
+      await viewerPage.expectUrlToContain('1.0');
+    });
+
+    it('[C586767] Previous document version title should be the same in Preview mode as the Uploaded File', async () => {
+      expect(await viewerPage.getFileTitle()).toContain(filesToUpload[0]);
+    });
+
+    it('[C586768] Should be possible to download a previous document version', async () => {
+      await viewerPage.clickDownloadButton();
+
+      expect(await Utils.fileExistsOnOS(filesToUpload[0])).toBe(true, 'File not found in download location');
+    });
+  });
+
+  describe('on Shared Files', () => {
+    beforeAll(async (done) => {
+      await loginPage.loginWith(username);
+      await page.clickSharedFilesAndWait();
```

Comment:
You need to share the file first, otherwise it won't be in this list

Reply:
Just saw you're sharing the file in the Personal Files suite. This creates a dependency between this suite and Personal Files. You could move line 39 in beforeAll of the main describe, or here.